### ISSUE TITLE
Quotes in paths/filenames

### ISSUE THREAD

**Issue by chrisism** [Wintermute0110/plugin.program.advanced.emulator.launcher#3]
Thought I just as well could create an issue instead of continueing in the pull request thread :)
But yeah, I found the quotes issue what I was talking about before. Here you see the case which gives some issues.

00:48:56 T:2048035048  NOTICE: AEL VERB : fs_load_ROM_XML_file() Loading XML file /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/userdata/addon_data/plugin.program.advanced.emulator.launcher/roms_Emulators_Megadrive_e0251.xml
00:48:58 T:2048035048  NOTICE: AEL INFO : _command_run_rom() application = "/system/xbin/su"
00:48:58 T:2048035048  NOTICE: AEL INFO : _command_run_rom() apppath     = "/system/xbin"
00:48:58 T:2048035048  NOTICE: AEL INFO : _command_run_rom() romfile     = "/sdcard/Shares/Media6/Roms/Sega - Mega Drive - Genesis/Zool - Ninja of the 'Nth' Dimension/Zool - Ninja of the 'Nth' Dimension (USA).md"
00:48:58 T:2048035048  NOTICE: AEL INFO : _command_run_rom() rompath     = "/sdcard/Shares/Media6/Roms/Sega - Mega Drive - Genesis/Zool - Ninja of the 'Nth' Dimension"
00:48:58 T:2048035048  NOTICE: AEL INFO : _command_run_rom() rombasename = "Zool - Ninja of the 'Nth' Dimension (USA).md"
00:48:58 T:2048035048  NOTICE: AEL INFO : _command_run_rom() arguments   = "-c 'am start -a android.intent.action.MAIN -c android.intent.category.LAUNCHER -e ROM "/sdcard/Shares/Media6/Roms/Sega - Mega Drive - Genesis/Zool - Ninja of the 'Nth' Dimension/Zool - Ninja of the 'Nth' Dimension (USA).md" -e LIBRETRO /data/data/com.retroarch/cores/genesis_plus_gx_libretro_android.so -n com.retroarch/.browser.retroactivity.RetroActivityFuture'"

Because on Android you can use the root su to start the emulator and you give it arguments like "-c 'am ... " you might get in trouble if the romfile already has quotes (like "..Ninja of the 'Nth' Dimension.."). 

I fixed this in the old AL for my case with some escaping. I already created a hotfix branch with the fix under https://github.com/chrisism/plugin.program.advanced.emulator.launcher/tree/hotfix/quote_escaping_with_arguments
But before we possibly merge it, I was thinking if the simple escaping might give issues in situations that its not needed. This fix works in my case because I already use quotes around it for the arguments. But how would it work on a system with a launcher that doesn't quote its arguments when launching a rom with quotes? So this should be tested before the fix can be merged. Don't want to break it on other systems.

**Comment by Wintermute0110** [comment]
I understand what you mean. Send your pull and I will merge. Will test in my box with Kodibuntu (shouldn't be any problem in Linux/OSX with your patch, I think). If some Windows user complains then I will add an option in the launcher "Quote ROM filename", similar to the former "Support lnk (Windows)".

In Linux I usually do "%rom%", so ROM name is already "" quoted. I think by chance I have not run a ROM with single quotes yet, but will try.

By the way, is it a good idea to run the emulator as root in Android?

**Comment by chrisism** [comment]
This evening I'll check my changes with my earlier ones on AL if its complete and then I'll create the pull request.

About running as root, I don't know if its due to the fact that I rooted the box. I tried the normal way (with /am) but it wouldnt start the emulator apps properly. Checked the kodi thread about it http://forum.kodi.tv/showthread.php?tid=183527. It could have also been that I was just screwing arround with the arguments too much.
But yeah if you use root or not, at least with this fix it shouldnt give any problems anymore. 
You'll see the pullrequest coming soon
